src/components/sidebar-toggle-button/SidebarToggleButton.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+import * as React from 'react';
+import classNames from 'classnames';
+import { injectIntl } from 'react-intl';
+
+import IconHide from '../../icons/general/IconHide';
+import IconShow from '../../icons/general/IconShow';
+import PlainButton from '../plain-button';
+import Tooltip from '../tooltip';
+
+import messages from '../../elements/common/messages';
+
+import './SidebarToggleButton.scss';
+
+type Props = {
+    isOpen: Boolean,
+    onClick?: Function,
+} & InjectIntlProvidedProps;
+
+const SidebarToggleButton = ({ intl, isOpen, onClick, ...rest }: Props) => {
```

Comment:
It looks like this component may not work for left-hand sidebars, since it expects the arrow and tooltip to face specific directions. If the goal is to make it a reusable component, should it take a `direction` prop? Should it also accept a `className`?